### PR TITLE
use one address allocator for all address allocations

### PR DESCRIPTION
**Pull request by NunoDasNeves**
Let ram_allocator be managed by system allocator, in a first step to unify address allocation to one place in the code.
TODO: let PCI segments be allocated from the system allocator.

https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4391

### CODE REVIEW

File: vm-allocator/src/system.rs [hunk 1]
```diff
@@ -123,9 +144,10 @@ impl SystemAllocator {
         size: GuestUsize,
         align_size: Option<GuestUsize>,
     ) -> Option<GuestAddress> {
-        self.platform_mmio_address_space.allocate(
+        self.ram_address_space.allocate_in_region(
```
Comment (rbradford) [inline vm-allocator/src/system.rs:147]: An important piece of nomenclature here. From our perspective RAM is memory in the address space that is backed by RAM pages from the host. MMIO memory, 32-bit or 64-bit is not RAM as there is no backing of it to RAM pages. `virtio-pmem` is backed by pages but those are not stored in RAM on the host (unless you `mmap()` a file in /tmp)

**Review by sboeuf — COMMENTED** [review]
This is very sensitive to bring changes to this area of the code.
I suggest you try to reach a point where you can both:
- show that you simplified the code (we need a strong reason for accepting changes like that)
- show that you have a full solution (we must really ensure there is no regression with the full solution, not only a small bit)

I don't think we can start evaluating/merging only small pieces as I'm worried we might end up in a deadend, that's why it'd be better to prove you have something working properly (CI passing) before we can start reviewing your code.

File: vm-allocator/src/system.rs (hunk 1 above)
Comment (NunoDasNeves) [reply vm-allocator/src/system.rs:150]: good point! maybe phys_address_space makes more sense

**Comment by rbradford** [comment]
@NunoDasNeves Do you mind closing this PR? Opened PRs consume resources by being continually built as part of the Jenkins branch indexing (which we need to clean up old builds but has the side effect of rebuilding all PRs..)

To me it's quite obvious that this PR isn't designed for merging as the commit messages don't follow the usual style and basic clippy/rustfmt checks are failing. I think we should continue the discussion on the issue - where of course you can reference this branch.

**Comment by NunoDasNeves** [comment]
> @NunoDasNeves Do you mind closing this PR? Opened PRs consume resources by being continually built as part of the Jenkins branch indexing (which we need to clean up old builds but has the side effect of rebuilding all PRs..)
> 
> To me it's quite obvious that this PR isn't designed for merging as the commit messages don't follow the usual style and basic clippy/rustfmt checks are failing. I think we should continue the discussion on the issue - where of course you can reference this branch.

No worries, I've closed it for now.

**Pull request closed without merging**